utilities/tools/Set-ModuleReadMe.ps1

```diff
@@ -460,11 +525,82 @@ function Set-DeploymentExamplesSection {
+                $orderedJSONParameters = ''
+            }
+
+            $jsonExample = ([ordered]@{
+                    '$schema'      = 'https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#'
+                    contentVersion = '1.0.0.0'
+                    parameters     = (-not [String]::IsNullOrEmpty($orderedJSONParameters)) ? $orderedJSONParameters : @{}
+                } | ConvertTo-Json -Depth 99)
+
+            # Optional: Add comment where required & optional parameters start
+            # ----------------------------------------------------------------
+            if ($requiredParameterNames -is [string]) {
+                $requiredParameterNames = @($requiredParameterNames)
+            }
+
+            # If we have at least one required and one other parameter we want to add a comment
+            if ($requiredParameterNames.Count -ge 1 -and $orderedJSONParameters.Keys.Count -ge 2) {
+
+                $jsonExampleArray = $jsonExample -split '\n'
+
+                # Check where the 'last' required parameter is located in the example (and what its indent is)
+                $parameterToSplitAt = $requiredParameterNames[-1]
+                $parameterStartIndex = ($jsonExampleArray | Select-String '.*"parameters": \{.*' | ForEach-Object { $_.LineNumber - 1 })[0]
+                $requiredParameterIndent = ([regex]::Match($jsonExampleArray[($parameterStartIndex + 1)], '^(\s+).*')).Captures.Groups[1].Value.Length
+
+                # Add a comment where the required parameters start
+                $jsonExampleArray = $jsonExampleArray[0..$parameterStartIndex] + ('{0}// Required parameters' -f (' ' * $requiredParameterIndent)) + $jsonExampleArray[(($parameterStartIndex + 1) .. ($jsonExampleArray.Count))]
+
+                # Find the location if the last required parameter
+                $requiredParameterStartIndex = ($jsonExampleArray | Select-String "^[\s]{$requiredParameterIndent}`"$parameterToSplitAt`": \{.*" | ForEach-Object { $_.LineNumber - 1 })[0]
+
+                # If we have more than only required parameters, let's add a corresponding comment
+                if ($orderedJSONParameters.Keys.Count -gt $requiredParameterNames.Count ) {
+                    # Search in rest of array for the next closing bracket with the same indent - and then add the search index (1) & initial index (1) count back in
+                    $requiredParameterEndIndex = ($jsonExampleArray[($requiredParameterStartIndex + 1)..($jsonExampleArray.Count)] | Select-String "^[\s]{$requiredParameterIndent}\}" | ForEach-Object { $_.LineNumber - 1 })[0] + 1 + $requiredParameterStartIndex
+
+                    # Add a comment where the non-required parameters start
+                    $jsonExampleArray = $jsonExampleArray[0..$requiredParameterEndIndex] + ('{0}// Non-required parameters' -f (' ' * $requiredParameterIndent)) + $jsonExampleArray[(($requiredParameterEndIndex + 1) .. ($jsonExampleArray.Count))]
```

Comment:
What about `additional` instead of `non-required`?
```suggestion
                    # Add a comment where the additional parameters start
                    $jsonExampleArray = $jsonExampleArray[0..$requiredParameterEndIndex] + ('{0}// Additional parameters' -f (' ' * $requiredParameterIndent)) + $jsonExampleArray[(($requiredParameterEndIndex + 1) .. ($jsonExampleArray.Count))]
```